### PR TITLE
fix: limit number of particles using the eventinfo

### PR DESCRIPTION
**Pull request by jhgoh**
`read_lhe()` tries to append to the list of particles in the event, raising error for some LHE output with optional fields. Another function `read_lhe_with_attributes()` is safe from this issue because it skips strings starting with '#'.
From the LHE standard, the number of particles is already given in the first line of the LHE event, which is also available in the eventinfo.nparticles, and any lines after nparticles should be treated as optional field.

Related issue: #102

### CODE REVIEW

**Review by matthewfeickert — CHANGES_REQUESTED** [review]
As @eduardo-rodrigues had pointed out in Issue #102, it would be good to cover this with tests. I understand that as this has gotten ignored for a long time that this might not be something @jhgoh has time to add now (totally understandable and reasonable given the delay), so if that's the case then once I'm through the next week or so I can come and add tests for all the PRs that are lacking them.

**Comment by eduardo-rodrigues** [comment]
Hi @matthewfeickert, I actually tested locally that this MR is all good with a file containing those lines starting with `#aMCatNLO`, as I found a file online for that. And since all tests present work, this MR is good to go. At this point in time a new release is really in need, and this MR is outstanding since long. I would kindly suggest that a new big release is made and more tests are added for robustness. I'm happy to add the file I'm mentioning to `scikit-hep-testdata` for a follow-up MR. WDYT?

**Comment by matthewfeickert** [comment]
> I'm happy to add the file I'm mentioning to `scikit-hep-testdata` for a follow-up MR.

Yeah a PR that adds more test data to `scikit-hep-testdata` would be great.

**Comment by eduardo-rodrigues** [comment]
> > I'm happy to add the file I'm mentioning to `scikit-hep-testdata` for a follow-up MR.
> 
> Yeah a PR that adds more test data to `scikit-hep-testdata` would be great.

@matthewfeickert, I have added a test based on the file I added to `scikit-hep-testdata`. This is ready to be merged IMHO. Are you happy at this stage? [Test coverage is something to improve but unrelated to this MR.]

**Comment by eduardo-rodrigues** [comment]
Closes https://github.com/scikit-hep/pylhe/issues/102

**Comment by eduardo-rodrigues** [comment]
Hello @matthewfeickert, @lukasheinrich, @kratsg, I have added the requested test some time ago. Anything else or this can go in :-)?

File: src/pylhe/__init__.py [hunk 1]
```diff
@@ -288,9 +288,10 @@ def read_lhe(filepath):
         with _extract_fileobj(filepath) as fileobj:
             for event, element in ET.iterparse(fileobj, events=["end"]):
                 if element.tag == "event":
-                    data = element.text.split("\n")[1:-1]
+                    data = element.text.strip().split("\n")
```
Comment (matthewfeickert) [inline src/pylhe/__init__.py:291]: @eduardo-rodrigues to answer your question on if this is good to merge, these two sections are uncovered still. It is late for me at the moment, but I can look at adding tests tomorrow unless you beat me to it. (I'm catching up on a backlog of work from holiday and being sick).

File: src/pylhe/__init__.py [hunk 2]
```diff
@@ -288,9 +288,10 @@ def read_lhe(filepath):
...
                     eventdata, particles = data[0], data[1:]
                     eventinfo = LHEEventInfo.fromstring(eventdata)
+                    particles = particles[: int(eventinfo.nparticles)]
```
Comment (matthewfeickert) [inline src/pylhe/__init__.py:294]: Same wrt coverage.

File: src/pylhe/__init__.py (hunk 1 above)
Comment (eduardo-rodrigues) [reply src/pylhe/__init__.py:291]: Sorry to hear that. Hope nothing too serious.

Seen you've changed job. Congrats!

Comment (eduardo-rodrigues) [reply src/pylhe/__init__.py:291]: [Today is going to be difficult for me to do developments.]

Comment (eduardo-rodrigues) [reply src/pylhe/__init__.py:291]: How do you want to test this trivial line `data = element.text.strip().split("\n")`, @matthewfeickert ?

Comment (eduardo-rodrigues) [reply src/pylhe/__init__.py:286]: Will try and look into this to finalise asap ... For the moment I've made this branch in sync with master, as it had diverged quite a bit.

Comment (eduardo-rodrigues) [reply src/pylhe/__init__.py:286]: Done.

File: src/pylhe/__init__.py (hunk 2 above)
Comment (eduardo-rodrigues) [reply src/pylhe/__init__.py:289]: Test coverage fixed here.

**Comment by eduardo-rodrigues** [comment]
Done the homework @matthewfeickert. I would be happy with an approval at this stage so that I merge :-).

**Comment by eduardo-rodrigues** [comment]
Morning @matthewfeickert, let's try and tick this off for the weekend. At this point is seems straightforward since I trivially dealt with your 2 remaining comments.

**Comment by eduardo-rodrigues** [comment]
Am now merging for the first time in this repo :-). I'm sure I ticked all to-dos from the review.

**Pull request merged**